### PR TITLE
Fixed spmd typecheck for qwen35 and kimi27

### PR DESCRIPTION
**Pull request by jinsooihm**
There were some changes in #4121 that made the spmd typecheck fail. 
The integration tests do not test for spmd_typecheck with TP > 1 for qwen35 and kimi27, so this was not caught. 

## Test plan

- Qwen3.5 multimodal, DP=4/TP=2, `spmd_types` strict typechecking: 1 training step passed.
- Kimi K2.5 multimodal, DP=2/TP=2, `spmd_types` strict typechecking: 1 training step passed using a test-only AdamW configuration with the TP-incompatible QK-clip hook disabled.
- `scripts/loss_compare.py`: compared parent `4ab7596c0` against `c302e9fb1` on Qwen3.5 TP=2 for 10 deterministic steps. Full-precision loss and `grad_norm` were bitwise identical at every step.
- Focused `ufmt`, `flake8`, `pydoclint`, and `git diff --check` passed.

### CODE REVIEW

**Review by tianyu-l — APPROVED** [review]

**Comment by tianyu-l** [comment]
> The integration tests do not test for spmd_typecheck with TP > 1

should add TP

**Comment by jinsooihm** [comment]
I added TP for qwen35. Kimi does not officially support TP yet because of the optimizer, so a test for kimi was not added.

File: torchtitan/models/qwen3_5/vision_encoder.py [hunk 1]
```diff
@@ -212,7 +214,9 @@ def _compute_2d_rope_cache(
     packed_rope_embeds = torch.cat([rope_embeds[i] for i in range(len(grids))], dim=0)
     if get_spmd_backend() == "spmd_types" and spmd.is_type_checking():
         packed_rope_embeds = spmd.mutate_type(
-            packed_rope_embeds, src=spmd.R, dst={"dp": spmd.V, "tp": spmd.I}
+            packed_rope_embeds,
+            src={"dp": spmd.R, "tp": spmd.I},
+            dst={"dp": spmd.V, "tp": spmd.I},
```
Comment (pianpwk) [inline torchtitan/models/qwen3_5/vision_encoder.py]: nit: you can change these calls to `"dp", src=R, dst=V`, but why weren't they starting as R@TP ? I thought factory methods would be R

**Review by pianpwk — APPROVED** [review]

File: torchtitan/models/qwen3_5/vision_encoder.py (hunk 1 above)
Comment (jinsooihm) [reply torchtitan/models/qwen3_5/vision_encoder.py]: I think before, these tensors were made from `torch.zeros`, but later PR changed them to be constructed from I@TP tensor and the type propagated.
